lib/hdiutil.js

```diff
@@ -43,20 +44,24 @@ exports.create = function (volname, size, cb) {
 }
 
 exports.attach = function (path, cb) {
+  var mountpoint = ""
+  do {
+    mountpoint = randomPath('/Volumes', '%s')
+  } while (fs.existsSync(mountpoint))
+
   const args = [
     'attach', path,
     '-nobrowse',
     '-noverify',
-    '-noautoopen'
+    '-noautoopen',
+    '-mountpoint',
+    mountpoint
   ]
 
   util.sh('hdiutil', args, function (err, res) {
     if (err) return cb(err)
 
-    const m = /Apple_HFS\s+(.*)\s*$/.exec(res.stdout)
-    if (m === null) return cb(new Error('Failed to mount image'))
-
-    cb(null, m[1])
+    cb(null, mountpoint)
```

Comment:
Hmm, actually, I think that this will create problems. The mount point is passed to the `bless` command later, and I think that it needs to match what will be the name on the target computer when they open the DMG file.

I think this because I know that when the process have been interrupted and left one attached disk, the next one gets ` (1)` appended to the name. And when that happens, I _think_, that the final dmg doesn't open automatically when mounted on other computers.

But the `openfolder` flag isn't supported at all on Apple Silicon Macs, so it might not be used on newer versions of macOS.

In order to preserve compatibility, what do you think about simply updating the regex so that it handles APFS as well? Then we could potentially drop the `openfolder` flag altogether in the next major version.

Sorry for not realising this before asking you to change the random thing 🙈

Reply:
Hmm, okay the output is a bit more complicated:

```text
❯ hdiutil attach /private/tmp/test-apfs-1.dmg
/dev/disk5          	GUID_partition_scheme          	
/dev/disk5s1        	Apple_APFS                     	
/dev/disk6          	EF57347C-0000-11AA-AA11-0030654	
/dev/disk6s1        	41504653-0000-11AA-AA11-0030654	/Volumes/Test APFS 1
```

vs.

```text
❯ hdiutil attach /private/tmp/test-hfsp-1.dmg   
/dev/disk7          	GUID_partition_scheme          	
/dev/disk7s1        	Apple_HFS                      	/Volumes/Test HFS+ 1
```

I think that this should work:

```js
/\s+(\/Volumes\/.*)\b/
```